### PR TITLE
Actor timer race condition

### PR DESCRIPTION
**Pull request by vinayada1**
# Description

Fixed issue with firing of timers and a race condition

#1871 

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1871 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/actors/actors.go [hunk 1]
```diff
@@ -1042,6 +1046,20 @@ func (a *actorsRuntime) CreateTimer(ctx context.Context, req *CreateTimerRequest
...
+		// Check if timer is still active
+		select {
+		case <-stop:
+			log.Infof("Time: %v with parameters: DueTime: %v, Period: %v, Data: %v has been deleted.", timerKey, req.DueTime, req.Period, req.Data)
+			return
+		default:
+			break
+		}
+
+		err := a.executeTimer(actorType, actorID, name, dueTime, period, callback, data)
```
Comment (youngbupark) [inline pkg/actors/actors.go:1058]: Why do we need call executeTimer here?

Comment (vinayada1) [reply pkg/actors/actors.go:1058]: Else, the timer is not firing after dueTime...it fires only after the period interval is over

Comment (youngbupark) [reply pkg/actors/actors.go:1058]: Sorry for the late. How did it work before?

Comment (vinayada1) [reply pkg/actors/actors.go:1058]: I dont think it worked

**Pull request merged**